src/rust/bin/wallet_tool.rs

```diff
@@ -0,0 +1,580 @@
+            format!("{}r{}", base, r)
+        } else {
+            base.to_string()
+        };
+        println!(
+            concat!(
+                "\nEnter the filename for the wallet export file, using only characters\n",
+                "a-z, A-Z and 0-9 (default '{}')."
+            ),
+            default_filename
+        );
+        let mut buf = String::new();
+        let response = prompt(&mut stdin, &mut buf)?;
+        let filename = if response.is_empty() {
+            r = r.saturating_add(1);
+            &default_filename
+        } else {
+            response
+        };
+        if opts.debug {
+            eprintln!("DEBUG: Using filename {:?}", filename);
+        }
+
+        let mut cli_args = cli_options.clone();
+        cli_args.extend_from_slice(&["z_exportwallet".to_string(), filename.to_string()]);
+        let out = exec(&zcash_cli, &cli_args, opts.debug)?;
+        let cli_err: Vec<_> = from_utf8(&out.stderr)
+            .with_context(|| "Output from zcash-cli was not UTF-8")?
+            .lines()
+            .map(|s| s.trim_end_matches('\r'))
+            .collect();
+        if opts.debug {
+            eprintln!("DEBUG: stderr {:?}", cli_err);
+        }
+        if cli_err.len() >= 3
+            && cli_err[0] == "error code: -8"
+            && cli_err[1] == "error message:"
+            && cli_err[2].contains("overwrite existing")
+        {
+            println!(concat!(
+                "That file already exists. Please pick a unique filename in the\n",
+                "directory specified by the '-exportdir' option to zcashd."
+            ));
+            continue;
+        } else {
+            break out;
+        }
+    };
+
+    let cli_out: Vec<_> = from_utf8(&out.stdout)
+        .with_context(|| "Output from zcash-cli was not UTF-8")?
+        .lines()
+        .map(|s| s.trim_end_matches('\r'))
+        .collect();
+    if opts.debug {
+        eprintln!("DEBUG: stdout {:?}", cli_out);
+    }
+    if cli_out.is_empty() {
+        return Err(WalletToolError::UnexpectedResponse.into());
+    }
+    let export_path = cli_out[0];
+
+    println!("\nSaved the export file to '{}'.", export_path);
+    println!("IMPORTANT: This file contains secrets that allow spending all wallet funds.\n");
+
+    let export_file = File::open(export_path)
+        .with_context(|| format!("Could not open {:?} for reading", export_path))?;
+    let phrase_line: Vec<_> = io::BufReader::new(export_file)
+        .lines()
+        .filter(|s| {
+            s.as_ref()
+                .map(|t| t.starts_with("# - recovery_phrase=\""))
+                .unwrap_or(false)
+        })
+        .collect();
+    if phrase_line.len() != 1 || phrase_line[0].is_err() {
+        return Err(WalletToolError::RecoveryPhraseNotFound.into());
+    }
+    let phrase = phrase_line[0]
+        .as_ref()
+        .unwrap()
+        .trim_start_matches("# - recovery_phrase=\"")
+        .trim_end_matches('"');
```

Comment:
We can avoid the explicit indexing and unwrapping:
```suggestion
    let phrase = match &phrase_line[..] {
        [Ok(line)] => line
            .expose_secret()
            .trim_start_matches("# - recovery_phrase=\"")
            .trim_end_matches('"'),
        _ => return Err(WalletToolError::RecoveryPhraseNotFound.into()),
    };
```